### PR TITLE
Ability to play media on canonical for CPS and legacy MAPs

### LINKED ISSUE #5919: Media Playback on Legacy MAPs in E2Es
Enable the 'play media' Cypress tests for Legacy MAPs

### PR DESCRIPTION
**Pull request by karinathomasbbc**
Resolves #5919  

**Overall change:** 
Updating the logic to allow media playback on legacy MAPs

**Code changes:**
- Add helper functions
- Update logic to play media on Canonical, and to check the iframe source on both AMP and Canonical

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added labels to this PR for the relevant pod(s) affected by these changes
- [ ] I have assigned this PR to the Simorgh project

**Testing:**
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

File: cypress/integration/pages/mediaAssetPage/helpers.js [hunk 1]
```diff
@@ -14,3 +16,29 @@ export const getBlockData = (blockType, body) => {
...
+
+export const getEmbedUrl = (body, language) => {
+  const isLegacyMedia = body.promo.media.type === 'legacyMedia';
+  const prefix = isLegacyMedia ? 'legacy' : 'cps';
+
+  return [
+    envConfig.avEmbedBaseUrl,
+    'ws/av-embeds',
+    `${prefix}${body.metadata.locators.assetUri}`,
+    getServiceId(body),
```
Comment (ryanmccombe) [inline cypress/integration/pages/mediaAssetPage/helpers.js]: This part of the URL shouldn't be the service - it should be:
- the vpid for modern maps
- the block id for legacy maps

Comment (ryanmccombe) [reply cypress/integration/pages/mediaAssetPage/helpers.js]: actually the function does look like it's returning the correct thing - it's just misnamed :)

**Review by ryanmccombe — APPROVED** [review]
👍

File: cypress/integration/pages/mediaAssetPage/helpers.js [hunk 2]
```diff
@@ -14,3 +16,29 @@ export const getBlockData = (blockType, body) => {
...
 };
+
+export const hasMedia = body => {
+  const mediaTypes = ['video', 'version', 'media', 'legacyMedia'];
+  return mediaTypes.some(type => body.metadata.blockTypes.includes(type));
+};
+
+const getServiceId = body => {
+  const mediaBlock = body.promo.media;
+  const versionId = mediaBlock.versions && mediaBlock.versions[0].versionId;
```
Comment (ryanmccombe) [inline cypress/integration/pages/mediaAssetPage/helpers.js]: Ramda has a function that might simplify this logic: https://ramdajs.com/docs/#paths

Something like:
```
const [vpid, externalId, id] = paths([
  ['versions', '0', 'versionId'],
  ['externalId'],
  ['id']
], mediaBlock)

return vpid || externalId || id;
```

**Review by pharingee — APPROVED** [review]
LGTM

**Pull request merged**